packages/core/modules-sdk/src/link.ts

```diff
@@ -403,15 +479,34 @@ export class Link {
         fields.push(link.data)
       }
 
-      serviceLinks.get(service.__definition.key)?.push(fields as any)
+      serviceLinks
+        .get(service.__definition.key)
+        ?.linksToCreate.push(fields as any)
     }
 
-    const promises: Promise<unknown[]>[] = []
+    for (const [serviceName, data] of serviceLinks) {
+      if (data.linksToValidateForUniqueness.filters.length) {
+        const service = this.modulesMap.get(serviceName)!
+        const existingLinks = await service.count({
```

Comment:
suggestion: Instead of implementing the new method `count`, maybe we should just use the method `list` with a `take: 1`, as the number of rows here is not important.

Reply:
Updated usage to `list`. Will remove the custom count method after testing the snapshot